stm32: drivers: adc: fix adc acquisition time comparison in adc_stm32_sampling_time_check

kbidani opened the pull request:
Add an explicit check to ensure that the acquisition_time parameter is encoded with the ADC_ACQ_TIME macro and uses the TICKS unit, as required by the API.
If the unit is not correct, log an error and return -EINVAL.

fixes: #94071

etienne-lms commented on drivers/adc/adc_stm32.c:
```diff
@@ -1211,8 +1211,7 @@ static int adc_stm32_sampling_time_check(const struct device *dev, uint16_t acq_
 	for (int i = 0; i < STM32_NB_SAMPLING_TIME; i++) {
-		if (acq_time == ADC_ACQ_TIME(ADC_ACQ_TIME_TICKS,
-					     config->sampling_time_table[i])) {
+		if (acq_time == config->sampling_time_table[i]) {
```
FYI I've posted a [comment in 94071](https://github.com/zephyrproject-rtos/zephyr/issues/94071#issuecomment-3187605445) to clarify the issue.

kbidani committed:
drivers: adc: stm32: Add unit check for acquisition_time

Add an explicit check to ensure that the acquisition_time
parameter is encoded with the ADC_ACQ_TIME macro and uses
the TICKS unit, as required by the API.
If the unit is not correct, log an error and return -EINVAL.

Signed-off-by: Khaoula Bidani <khaoula.bidani-ext@st.com>

etienne-lms commented:
I'm adding a *DNM* label since I don't think this change is valid, see https://github.com/zephyrproject-rtos/zephyr/issues/94071#issuecomment-3187605445.
